information/specification.md

```diff
@@ -0,0 +1,826 @@
+    coin value_.
+
+    Entries are incrementally removed from the _head_ of the list as the
+    algorithm proceeds, until the list is empty.
+
+ 2. #### Unpaid Output List
+
+    This is initially equal to the [requested output
+    list](#requested-output-list), sorted into _descending order of coin
+    value_.
+
+    The _head_ of the list is always the unpaid output with the _largest coin
+    value_.
+
+    Entries are incrementally removed from the _head_ of the list as the
+    algorithm proceeds, until the list is empty.
+
+ 3. #### Accumulated Coin Selection
+
+    The accumulated coin selection is a [coin selection](#coin-selection) where
+    all fields are initially equal to the _empty set_.
+
+### Computation
+
+The algorithm proceeds according to the following sequence of steps:
+
+ * **Step 1**
+
+   Remove a single _unpaid output_ from the head of the [unpaid output
+   list](#unpaid-output-list).
+
+ * **Step 2**
+
+   Repeatedly remove UTxO entries from the head of the [available UTxO
+   list](#available-utxo-list) until the total value of entries removed is
+   _greater than or equal to_ the value of the _removed output_.
+
+ * **Step 3**
+
+   Use the _removed UTxO entries_ to pay for the _removed output_.
+
+   This is achieved by:
+
+    * adding the _removed UTxO entries_ to the _inputs_ field of the
+      [accumulated coin selection](#accumulated-coin-selection).
+    * adding the _removed output_ to the _outputs_ field of the
+      [accumulated coin selection](#accumulated-coin-selection).
+
+ * **Step 4**
+
+   If the _total value_ of the _removed UTxO entries_ is greater than the
+   value of the _removed output_, generate a coin whose value is equal to
+   the exact difference, and add it to the _change values_ field of the
+   [accumulated coin selection](#accumulated-coin-selection).
+
+ * **Step 5**
+
+   If the _unpaid output list_ is empty, **terminate** here.
+
+   Otherwise, return to **Step 1**.
+
+### Termination
+
+The algorithm terminates _successfully_ if the [available UTxO
+list](#available-utxo-list) is not depleted before the [unpaid output
+list](#unpaid-output-list) can be fully depleted (i.e., if all the outputs have
+been paid for).
+
+The [accumulated coin selection](#accumulated-coin-selection) is returned
+to the caller as the [coin selection](#coin-selection) result.
+
+The [available UTxO list](#available-utxo-list) is returned to the caller
+as the [remaining UTxO set](#remaining-utxo-set) result.
+
+## Random-Improve
+
+The **Random-Improve** coin selection algorithm works in _two phases_.
+
+In the _first_ phase, the algorithm iterates through each of the [requested
+outputs](#requested-output-list) in _descending order of coin value_, from
+_largest_ to _smallest_. For each output, the algorithm repeatedly selects
+entries at **random** from the [initial UTxO set](#initial-utxo-set), until the
+_total value_ of selected entries is enough to pay for that ouput.
```

Comment:
for that _ouput_ → for that _ou**t**put_

Reply:
> for that _ouput_ → for that _ou**t**put_

Fixed in e252dd3c33f9a076c2a08a3ed0d74a494c491880.